File: 0787. Cheapest Flights Within K Stops/Solution0787.py
class Solution0787:
    def findCheapestPrice(self, n: int, flights: List[List[int]], src: int, dst: int, k: int) -> int:
        dp = [float('inf')] * n
        dp[src] = 0
        for i in range(k+1):
            prev = dp.copy()
            for flight in flights:
                dp[flight[1]] = min(dp[flight[1]], prev[flight[0]] + flight[2])
        return dp[dst] if dp[dst] != float('inf') else -1

File: 0787. Cheapest Flights Within K Stops/Solution0787.java
class Solution0787 { // Bellman-ford
    public int findCheapestPrice(int n, int[][] flights, int src, int dst, int k) {
        int[] dp = new int[n];
        int INF = Integer.MAX_VALUE / 2;
        Arrays.fill(dp, INF);
        dp[src] = 0;
        for (int i = 0; i < k + 1; i++) {
            int[] prev = dp.clone();
            for (int[] flight : flights) {
                dp[flight[1]] = Math.min(dp[flight[1]], prev[flight[0]] + flight[2]);
            }
        }
        return dp[dst] == INF ? -1 : dp[dst];
    }
}